### PR TITLE
Certificate handling improvements for preview environments

### PR DESCRIPTION
**Pull request by mads-hartmann**
## Description

We have seen sporadic problem when issuing certificate for our preview environments. It appears to be a problem with CertManager and that upgrading to a newer version might reduce the chances of getting into this problem, but from reports it doesn't seem to eliminate it completely.

This PR:

1. Makes it easier for developers to know how to proceed when certificates fail. We delete the certificate and instruct them to re-run their job
2. Helps the platform team more proactively debug and fix these issues by tagging us in a Slack message and pointing us to a [new internal guide](https://www.notion.so/gitpod/Debugging-certificate-issues-9453d1c8ac914ce7962557b67f7b49b3) we can improve over time. 

Additionally the PR improves a few things

1. Use werft.fail instead of throwing errors directly. This improves the werft logs and the underlying trace while still resulting in an error being thrown.
2. Removing superfluous try/catch/throw error code
3. Fixes some (I believe) unwanted parallelism - see comments directly in the code.

## Related Issue(s)

Fixes https://github.com/gitpod-io/ops/issues/2066

## How to test

I haven't been able to validate the error-handling code for when the certificates fail due to the nature of the issues. 

## Release Notes

```release-note
NONE
```

## Documentation

N/A

### CODE REVIEW

File: .werft/jobs/build/deploy-to-preview-environment.ts [hunk 1]
```diff
@@ -142,7 +142,7 @@ export async function deployToPreviewEnvironment(werft: Werft, jobConfig: JobCon
 
         try {
             werft.log(vmSlices.COPY_CERT_MANAGER_RESOURCES, 'Copy over CertManager resources from core-dev')
-            installMetaCertificates(werft, jobConfig.repository.branch, withVM, 'default', PREVIEW_K3S_KUBECONFIG_PATH, vmSlices.COPY_CERT_MANAGER_RESOURCES)
```
Comment (mads-hartmann) [inline .werft/jobs/build/deploy-to-preview-environment.ts:145]: I believe we want this to be blocking and not fire-and-forget. The certificates has previously been issues earlier in the job, so this is waiting for them to be ready and then copying them over.

File: .werft/jobs/build/prepare.ts [hunk 2]
```diff
@@ -24,7 +23,7 @@ export async function prepare(werft: Werft, config: JobConfig) {
         configureStaticClustersAccess()
         werft.done(prepareSlices.CONFIGURE_CORE_DEV)
 
-        issueCertificate(werft, config)
+        await issueCertificate(werft, config)
```
Comment (mads-hartmann) [inline .werft/jobs/build/prepare.ts:26]: This is just creating the Certificate CRD resources that  CertManager uses. I don't believe we want to do so as a "fire-and-forget" promise

File: .werft/jobs/build/prepare.ts [hunk 3]
```diff
@@ -62,12 +61,12 @@ function configureStaticClustersAccess() {
...
 }
 
-function issueCertificate(werft: Werft, config: JobConfig) {
+async function issueCertificate(werft: Werft, config: JobConfig) {
     const certName = config.withVM ? `harvester-${previewNameFromBranchName(config.repository.branch)}` : `staging-${previewNameFromBranchName(config.repository.branch)}`
     const domain = config.withVM ? `${config.previewEnvironment.destname}.preview.gitpod-dev.com` : `${config.previewEnvironment.destname}.staging.gitpod-dev.com`;
 
     werft.log(prepareSlices.ISSUE_CERTIFICATES, prepareSlices.ISSUE_CERTIFICATES)
-    issueMetaCerts(werft, certName, "certs", domain, config.withVM, prepareSlices.ISSUE_CERTIFICATES)
+    await issueMetaCerts(werft, certName, "certs", domain, config.withVM, prepareSlices.ISSUE_CERTIFICATES)
```
Comment (mads-hartmann) [inline .werft/jobs/build/prepare.ts:69]: Same as the comment above, this is just creating the Certificate, no need to use "fire-and-forget"

File: .werft/jobs/build/deploy-to-preview-environment.ts (hunk 1 above)
Comment (meysholdt) [reply .werft/jobs/build/deploy-to-preview-environment.ts:145]: 👍

File: .werft/jobs/build/prepare.ts (hunk 2 above)
Comment (meysholdt) [reply .werft/jobs/build/prepare.ts:26]: agreed - assuming we only create the CR and don't actually wait for the cert to be issued.

File: .werft/jobs/build/prepare.ts (hunk 3 above)
Comment (meysholdt) [reply .werft/jobs/build/prepare.ts:69]: 👍

**Review by meysholdt — APPROVED** [review]
Code changes LGTM

**Pull request merged**